src/axom/quest/tests/quest_sampling_shaper.cpp

```diff
@@ -0,0 +1,868 @@
+
+    return *gf * vol_form;
+  }
+
+  /// Registers and allocates a volume fraction grid function within the datastore
+  mfem::GridFunction* registerVolFracGridFunction(const std::string& name,
+                                                  int vfOrder = 2)
+  {
+    SLIC_ASSERT(!m_dc.HasField(name));
+
+    auto& mesh = getMesh();
+    const int dim = mesh.Dimension();
+
+    // create grid function
+    auto* coll =
+      new mfem::L2_FECollection(vfOrder, dim, mfem::BasisType::Positive);
+    auto* fes = new mfem::FiniteElementSpace(&mesh, coll);
+    auto* vf = new mfem::GridFunction(fes);
+    vf->MakeOwner(coll);
+
+    // allocate grid function via sidre
+    const int sz = fes->GetVSize();
+    mfem::Vector v(m_dc.AllocNamedBuffer(name, sz)->getData(), sz);
+    vf->MakeRef(fes, v, 0);
+
+    // register grid function w/ data collection
+    m_dc.RegisterField(name, vf);
+
+    return vf;
+  }
+
+  /** 
+   * \brief Initializes the values of the DOFs of a volume fraction grid function
+   * using a provided lambda w/ parameters for the cell index, DOF position and cell attribute
+   * The signature of DOFInitializer is [](int idx, Point2D& pt, int attribute) -> double
+   */
+  template <typename DOFInitializer>
+  void initializeVolFracGridFunction(mfem::GridFunction* vf,
+                                     DOFInitializer&& dof_initializer)
+  {
+    auto& mesh = this->getMesh();
+    const int dim = mesh.Dimension();
+    const int NE = mesh.GetNE();
+
+    // Assume all elements have the same integration rule
+    const auto* fes = vf->FESpace();
+    auto* fe = fes->GetFE(0);
+    auto& ir = fe->GetNodes();
+    const int nq = ir.GetNPoints();
+
+    // Get positions of DOFs
+    mfem::DenseTensor pos_coef(dim, nq, NE);
+    {
+      const auto* geomFactors =
+        mesh.GetGeometricFactors(ir, mfem::GeometricFactors::COORDINATES);
+
+      // Rearrange positions
+      for(int i = 0; i < NE; ++i)
+      {
+        for(int j = 0; j < dim; ++j)
+        {
+          for(int k = 0; k < nq; ++k)
+          {
+            pos_coef(j, k, i) = geomFactors->X((i * nq * dim) + (j * nq) + k);
+          }
+        }
+      }
+    }
+
+    // Initialize volume fraction DOFs using passed in lambda based on cell index, DOF position and attribute
+    mfem::Vector res(nq);
+    mfem::Array<int> dofs;
+    for(int idx = 0; idx < NE; ++idx)
+    {
+      const int attr = mesh.GetAttribute(idx);
+
+      mfem::DenseMatrix& m = pos_coef(idx);
+      for(int p = 0; p < nq; ++p)
+      {
+        const Point2D pt(m.GetColumn(p), dim);
+        res(p) = dof_initializer(idx, pt, attr);
+      }
+
+      fes->GetElementDofs(idx, dofs);
+      vf->SetSubVector(dofs, res);
+    }
+  }
+
+  /// Helper to check integrated volume of a volume fraction grid fucntion
+  void checkExpectedVolumeFractions(const std::string& material_name,
+                                    double expected_volume,
+                                    double EPS = 1e-2)
```

Comment:
How did you determine this default value for `EPS`?

Reply:
For the testing, I'm using a really coarse mesh and relatively low linearization and quadrature order parameters, so I didn't expect the result to be that accurate compared to the analytic results (i.e. $\pi * (r^2_{outer} - r^2_{inner})$ for the annulus). I was only checking that the results are reasonably close.